File: care_well_backend/CareWell/CareWell.Domain/Salud/PersonaEstadoAnimo.cs
using CareWell.Domain.General;
using CareWell.Domain.Validadores;
using CareWell.Domain.ValueObjects.Salud;
using CareWell.Global.Exceptions;
using CareWell.Global.Mensajes;

namespace CareWell.Domain.Salud
{
    public class PersonaEstadoAnimo : BaseEntity
    {
        public virtual Persona Persona { get; private set; }

        public virtual DateTime FechaHora { get; private set; }

        public virtual EstadoAnimo EstadoAnimo { get; private set; }

        public virtual string? Observaciones { get; private set; }

        public virtual void Registrar(RegistrarEstadoAnimo registrarEstadoAnimo,
                                      IValidadorPermisoAccion validadorPermisoAccion)
        {
            if (registrarEstadoAnimo.Persona is null)
                throw new ValidacionDominioException(Mensajes.PersonaNoExiste);

            validadorPermisoAccion.ValidarVisualizacion(registrarEstadoAnimo.Persona, registrarEstadoAnimo.Colaborador);

            if (registrarEstadoAnimo.EstadoAnimo is null)
                throw new ValidacionDominioException(Mensajes.EstadoAnimoRequerido);

            this.Persona = registrarEstadoAnimo.Persona;
            this.FechaHora = DateTime.Now;
            this.EstadoAnimo = registrarEstadoAnimo.EstadoAnimo;
            this.Observaciones = registrarEstadoAnimo.Observaciones;
        }
    }
}

File: care_well_app/lib/domain/entities/health/persona_estado_animo.dart
import 'package:care_well_app/domain/entities/entities.dart';

class PersonaEstadoAnimo extends BaseEntity {
  final Persona persona;
  final DateTime fecha;
  final EstadoAnimo estado;
  final String? observaciones;

  const PersonaEstadoAnimo({
    required super.id,
    required this.persona,
    required this.fecha,
    required this.estado,
    this.observaciones,
  });

  @override
  PersonaEstadoAnimo copyWith({
    int? id,
    Persona? persona,
    DateTime? fecha,
    EstadoAnimo? estado,
    String? observaciones,
  }) {
    return PersonaEstadoAnimo(
      id: id ?? this.id,
      persona: persona ?? this.persona,
      fecha: fecha ?? this.fecha,
      estado: estado ?? this.estado,
      observaciones: observaciones ?? this.observaciones,
    );
  }
}
